dcos/dcos-ui
# [spike] DCOS-49324: Service group creation rejects names that should be valid

Improve error handling in the modal
to give the user more accurate information.

Closes https://jira.mesosphere.com/browse/DCOS-49324

## Testing
1. In the services tab, try to create a group with an invalid name, for example ".".
2. Verify that our validation error is shown (instead of the cryptic server message).
3. Try to create a group using the name of a group that already exists.
4. Verify that a comprehensible error is shown (instead of the cryptic server message).
5. Verify that the added tests make sense.

## Trade-offs
https://github.com/dcos/dcos-ui/pull/3897#pullrequestreview-239970699

## Dependencies
None.

## Screenshots
### Before
![Screenshot from 2019-05-21 13-56-03](https://user-images.githubusercontent.com/40791275/58090601-3f7bc200-7bd0-11e9-9811-704a209ffeab.png)
![Screenshot from 2019-05-21 13-56-20](https://user-images.githubusercontent.com/40791275/58090604-41de1c00-7bd0-11e9-8916-bfbb4973fd05.png)

### After
![Screenshot from 2019-05-21 11-45-06](https://user-images.githubusercontent.com/40791275/58090518-122f1400-7bd0-11e9-81da-953633aafcb4.png)
![Screenshot from 2019-05-21 11-45-19](https://user-images.githubusercontent.com/40791275/58090525-14916e00-7bd0-11e9-8361-4ec839b2030d.png)
